homeassistant/components/hyperion/light.py

```diff
@@ -25,104 +25,152 @@
+KEY_SUBCOMMAND = "subcommand"
+KEY_SUBSCRIBE = "subscribe"
+KEY_SUCCESS = "success"
+KEY_SWITCH_TO = "switchTo"
+KEY_STATE = "state"
+KEY_TOKEN = "token"
+KEY_UPDATE = "update"
+KEY_VALUE = "value"
+KEY_VISIBLE = "visible"
+
+# ComponentIDs from: https://docs.hyperion-project.org/en/json/Control.html#components-ids-explained
+KEY_COMPONENTID = "componentId"
+KEY_COMPONENTID_ALL = "ALL"
+KEY_COMPONENTID_COLOR = "COLOR"
+KEY_COMPONENTID_EFFECT = "EFFECT"
+
+KEY_COMPONENTID_EXTERNAL_SOURCES = ["BOBLIGHTSERVER", "GRABBER", "V4L"]
+KEY_COMPONENTID_LEDDEVICE = "LEDDEVICE"
+
+# As we want to preserve brightness control for effects (e.g. to reduce the
+# brightness for V4L), we need to persist the effect that is in flight, so
+# subsequent calls to turn_on will know the keep the effect enabled.
+# Unfortunately the Home Assistant UI does not easily expose a way to remove a
+# selected effect (there is no 'No Effect' option by default). Instead, we
+# create a new fake effect ("Solid") that is always selected by default for
+# showing a solid color. This is the same method used by WLED.
+KEY_EFFECT_SOLID = "Solid"
 
 DEFAULT_COLOR = [255, 255, 255]
+DEFAULT_BRIGHTNESS = 255
+DEFAULT_EFFECT = KEY_EFFECT_SOLID
 DEFAULT_NAME = "Hyperion"
+DEFAULT_ORIGIN = "Home Assistant"
 DEFAULT_PORT = 19444
 DEFAULT_PRIORITY = 128
 DEFAULT_HDMI_PRIORITY = 880
-DEFAULT_EFFECT_LIST = [
-    "HDMI",
-    "Cinema brighten lights",
-    "Cinema dim lights",
-    "Knight rider",
-    "Blue mood blobs",
-    "Cold mood blobs",
-    "Full color mood blobs",
-    "Green mood blobs",
-    "Red mood blobs",
-    "Warm mood blobs",
-    "Police Lights Single",
-    "Police Lights Solid",
-    "Rainbow mood",
-    "Rainbow swirl fast",
-    "Rainbow swirl",
-    "Random",
-    "Running dots",
-    "System Shutdown",
-    "Snake",
-    "Sparks Color",
-    "Sparks",
-    "Strobe blue",
-    "Strobe Raspbmc",
-    "Strobe white",
-    "Color traces",
-    "UDP multicast listener",
-    "UDP listener",
-    "X-Mas",
-]
-
+DEFAULT_INSTANCE = 0
+DEFAULT_CONNECTION_RETRY_DELAY = 30
+DEFAULT_CONNECTION_TIMEOUT_SECS = 5
 SUPPORT_HYPERION = SUPPORT_COLOR | SUPPORT_BRIGHTNESS | SUPPORT_EFFECT
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_HOST): cv.string,
-        vol.Required(CONF_PORT, default=DEFAULT_PORT): cv.port,
-        vol.Optional(CONF_DEFAULT_COLOR, default=DEFAULT_COLOR): vol.All(
-            list,
-            vol.Length(min=3, max=3),
-            [vol.All(vol.Coerce(int), vol.Range(min=0, max=255))],
-        ),
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Optional(CONF_PRIORITY, default=DEFAULT_PRIORITY): cv.positive_int,
-        vol.Optional(
-            CONF_HDMI_PRIORITY, default=DEFAULT_HDMI_PRIORITY
-        ): cv.positive_int,
-        vol.Optional(CONF_EFFECT_LIST, default=DEFAULT_EFFECT_LIST): vol.All(
-            cv.ensure_list, [cv.string]
-        ),
-    }
+PLATFORM_SCHEMA = vol.All(
```

Comment:
We no longer allow changes to YAML configuration for integration that communicate with a device or service.

For more information see:

- ADR-0007: <https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md#decision>
- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md>

This integration needs to be adjusted to use a configuration flow. To get started on that, see our developer documentation:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

Reply:
Thank you -- I will work on that. I'm also refactoring the network connection out of the integration and into a common library. These changes will take some time though, so the integration will thus need to remain non-functional (unless someone beats me to it).

Reply:
Quick update on this: I have made good progress on creating, and unittesting, a [hyperion-py library](https://github.com/dermotduffy/hyperion-py) that can manage the raw device communication. That should make this home-assistant code cleaner.

@frenck -- your advice sought, I can either:
   1. Have 1 PR that modifies this integration in place to use the new library, but without touching the YAML at all -- then follow that with another for the config flow that removes the YAML. The exact intent of the YAML configuration won't work after the initial PR -- but most of the functionality probably will for the common usecases. Given that the integration doesn't work at all for the current Hyperion, this feels like not a terrible move.
   1. OR: I can combine the refactor and the config flow into one larger PR (which is just more difficult to review).

Let me know which you think might be better. Thanks!